src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -235,7 +235,15 @@ internal virtual TypeEnumerator GetTypeEnumerator(Type type, string assemblyFile
                 {
                     if (discoveryOption == TestDataSourceDiscoveryOption.DuringDiscovery)
                     {
-                        if (DynamicDataAttached(sourceLevelParameters, test, tests))
+                        Lazy<TestMethodInfo?> testMethodInfo = GetTestMethodInfo(sourceLevelParameters, test);
+
+                        // Add fixture tests like AssemblyInitialize, AssemblyCleanup, ClassInitialize, ClassCleanup.
+                        if (MSTestSettings.CurrentSettings.FixturesEnabled && testMethodInfo.Value is not null)
```

Comment:
This is called foreach type, we may want to have assembly fixtures filled only once.

Reply:
Inner method will populate it only once for Assembly fixtures. There is a check for it.